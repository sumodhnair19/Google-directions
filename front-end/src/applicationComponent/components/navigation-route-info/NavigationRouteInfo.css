.route-info {
    padding: 16px;
}

.route-info-item {
    font-size: 1em;
}

.route-info-item-label {
    font-weight: 500;
}

.route-info-item-value {
    margin-left: 8px;
    font-weight: bold;
}
