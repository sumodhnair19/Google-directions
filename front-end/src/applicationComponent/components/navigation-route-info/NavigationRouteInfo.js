import React from 'react';

import './NavigationRouteInfo.css';
/**
 * @name NavigationRouteItems
 * @type {Component}
 * @description Route info item component
 * @param props are the passed values from parent to stateless component
 * @returns {JSX}
 */

const NavigationRouteItems = (props) => (
    <div className="route-info-item">
        <div className="route-info-item-label">{props.title}: </div>
        <div className="route-info-item-value"> {props.value}</div>
    </div>
);
/**
 * @name NavigationRouteInfo
 * @type {Component}
 * @description Route info component. Display the route params like distance and time
 * @param props are the passed values from stateless to stateless component
 * @returns {JSX}
 */
const NavigationRouteInfo = (props) => (
    <div className="route-info">
        <NavigationRouteItems title="Total distance" value={props.total_distance} />
        <NavigationRouteItems title="Total time" value={props.total_time} />
    </div>
);

export default NavigationRouteInfo;
