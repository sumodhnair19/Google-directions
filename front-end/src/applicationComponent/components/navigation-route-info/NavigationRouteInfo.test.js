import React from 'react';
import renderer from 'react-test-renderer';
import NavigationRouteInfo from './NavigationRouteInfo';

describe('Navigation Route Info Test', () => {
    it('render correctly', () => {
        const tree = renderer.create(<NavigationRouteInfo />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
