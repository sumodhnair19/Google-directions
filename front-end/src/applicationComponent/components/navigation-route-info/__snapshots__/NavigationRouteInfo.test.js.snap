// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Route Info Test render correctly 1`] = `
<div
  className="route-info"
>
  <div
    className="route-info-item"
  >
    <div
      className="route-info-item-label"
    >
      Total distance
      : 
    </div>
    <div
      className="route-info-item-value"
    >
       
    </div>
  </div>
  <div
    className="route-info-item"
  >
    <div
      className="route-info-item-label"
    >
      Total time
      : 
    </div>
    <div
      className="route-info-item-value"
    >
       
    </div>
  </div>
</div>
`;
